### PR TITLE
handle multi line strings in console overlay

### PR DESCRIPTION
**Pull request by jwunderl**
splits up lines so multi line strings fit nicely into the console overlay; e.g. this:

instead of this:

this was annoying with https://github.com/microsoft/pxt-common-packages/pull/886, as the pretty printing wasn't that pretty due to everything overlapping

Also add optional parameter to to ``.setVisible``, so that people can set the color if they want to

https://makecode.com/_1y3W2ceD8C3J

### CODE REVIEW

File: libs/game/console.ts [hunk 1]
```diff
@@ -15,31 +15,40 @@ namespace game.consoleOverlay {
...
+        if (col !== undefined)
+            consoleColor = col;
     }
 
     function listener(priority: ConsolePriority, text: string) {
         if (!consoleStrings)
             return;
-        else if (consoleStrings.length < consoleLines)
+
+        let newLineIndex = text.indexOf("\n");
```
Comment (pelikhan) [inline libs/game/console.ts]: why are we not using "split"?

File: libs/game/console.ts [hunk 2]
```diff
@@ -15,31 +15,40 @@ namespace game.consoleOverlay {
...
+        if (consoleStrings.length > MAX_CONSOLE_LINES) {
+            consoleStrings.splice(0, consoleStrings.length - MAX_CONSOLE_LINES);
         }
     }
 
     export function draw() {
         if (!consoleStrings || scene.systemMenu.isVisible()) return;
-        const height = consoleFont.charHeight + marginy ;
-        const top = 2 + (game.stats ? height : 0);
+        const lineHeight = consoleFont.charHeight + marginy;
```
Comment (pelikhan) [inline libs/game/console.ts]: unrelated change

File: libs/game/console.ts (hunk 1 above)
Comment (jwunderl) [reply libs/game/console.ts]: Oops, must have just blanked for a bit, I did this right before leaving for the day. fixed

File: libs/game/console.ts [hunk 3]
```diff
@@ -15,27 +15,29 @@ namespace game.consoleOverlay {
...
             return;
-        else if (consoleStrings.length < consoleLines)
-            consoleStrings.push(text);
-        else {
-            for (let i = 1; i < consoleStrings.length; ++i) {
-                consoleStrings[i - 1] = consoleStrings[i];
-            }
-            consoleStrings[consoleStrings.length - 1] = text;
+
+        text.split("\n")
```
Comment (pelikhan) [inline libs/game/console.ts]: is fast to do array.concat?

Comment (jwunderl) [reply libs/game/console.ts]: concat returns a new array, so it would have to push all the old strings too instead of just the new ones https://github.com/microsoft/pxt/blob/master/libs/pxt-common/pxt-helpers.ts#L193
